core/src/main/java/org/eclipse/hono/util/MessageHelper.java

```diff
@@ -293,20 +294,19 @@ public static boolean getXOptAppCorrelationId(final Message msg) {
      */
     public static JsonObject getJsonPayload(final Message msg) {
 
-        final String payload = getPayload(msg);
-        return (payload != null ? new JsonObject(payload) : null);
+        final Buffer buffer = getPayload(msg);
+        return buffer != null ? new JsonObject(buffer.toString(StandardCharsets.UTF_8)) : null;
```

Comment:
return buffer != null ? buffer.toJsonObject() : null;

Reply:
Thank you - made changes.